### PR TITLE
check wallet activated when sending to stellar

### PR DESCRIPTION
**Pull request by AlaaElattar**
- https://github.com/threefoldtech/threefold_connect/issues/767

### Tested Scenarios:
- Tried to send to wallet with no stellar but the error appeared.

### CODE REVIEW

File: app/lib/screens/wallets/send.dart [hunk 1]
```diff
@@ -114,6 +114,15 @@ class _WalletSendScreenState extends State<WalletSendScreen> {
...
       }
+
+      final matchingWallets = widget.allWallets
+          .where((wallet) => wallet.stellarAddress == toAddress);
+      final Wallet? wallet =
+          matchingWallets.isNotEmpty ? matchingWallets.first : null;
+      if (wallet != null && wallet.stellarBalance == '-1') {
+        toAddressError = 'Wallet not activated on stellar';
+        return false;
+      }
```
Comment (zaelgohary) [inline app/lib/screens/wallets/send.dart]:
```suggestion
      final wallet = widget.allWallets
          .where((wallet) => wallet.stellarAddress == toAddress).toList().first;
      if (wallet.stellarBalance == '-1') {
        toAddressError = 'Wallet not activated on stellar';
        return false;
      }
```

Comment (AhmedHanafy725) [inline app/lib/screens/wallets/send.dart]: not enough, what if the user copies and pastes an address that does not match the current wallets?

we should get the balance of the wallet and make sure it's activated and has TFT trustline

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: app/lib/screens/wallets/send.dart (hunk 1 above)
Comment (AlaaElattar) [reply app/lib/screens/wallets/send.dart]: In the provided code if there's no matching wallet, flutter will throw an exception.
That's why i handled it that way to check first there's a matching wallet.

Comment (AlaaElattar) [reply app/lib/screens/wallets/send.dart]: should be fixed

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**